templates/tpus.libsonnet

```diff
@@ -23,24 +25,34 @@
     replicas: tpu.size / 8, # Each TPU replica has 8 cores
     preemptible: false,
 
-    PodSpec:: {
-      containerMap+: {
-        train+: {
-          resources+: {
-            limits+: { [tpu.resource]: tpu.size },
+    PodTemplate(tpuSettings):: {
+      metadata: {
+        annotations: {
+          "tf-version.cloud-tpus.google.com": tpuSettings.softwareVersion,
+        },
+      },
+      spec+: {
+        containerMap+: {
+          train+: {
+            resources+: {
+              local preemptiblePrefix = 
+                if tpuSettings.preemptible then
+                  "preemptible-"
+                else
+                  "",
+              local resourceName = "cloud-tpus.google.com/%sv%s" % [preemptiblePrefix, tpu.version],
+
+              limits+: { [resourceName]: tpu.size },
+            },
           },
         },
+        nodeSelector+:
+          if tpuSettings.requireTpuAvailableLabel then
```

Comment:
Do you need to set this to True somewhere? Looks like default is False in the base template

Reply:
This is set to true in `tests/common.libsonnet`. This selector only makes sense in the context of a regional cluster, which is going to be overkill for most teams, so the default is false.